src/Services/CharacterManagerService.php

```diff
@@ -16,14 +17,29 @@
 class CharacterManagerService extends BaseManagerService
 {
     /**
-     * Creates a new character
+     * Creates a new character.
      * @param string $name
      * @return Character
+     * @throws ArgumentException If character name is invalid
      * @throws CharacterNameExistsException If character is already taken
-     * @throws CrateException If some unknown character occured during saving.
+     * @throws CrateException If some unknown character occured during saving
      */
     public function createNewCharacter(string $name): Character
     {
+        // Remove whitespace
+        $name = trim($name);
+
+        if (empty($name)) {
+            throw new ArgumentException("Character name must not be empty.");
+        }
+
+        // Remove special characters and check if it is still the same name
+        $name_sanitized = preg_replace("/[^\\p{L} ]/u", "", $name);
```

Comment:
I dont think this restriction is a good idea.

There are plenty of reasonable characters this excludes, like '",|()*^@... For sure we should allow some of those if not all symbols.

It also doesnt allow any emoji, which I think a modern game should allow. Can you dig around for a regexp that covers most reasonable "character-like" things? So everything but control characters I suppose, like ^A etc.